docs/using-availability-zones.md

```diff
@@ -0,0 +1,220 @@
+# Availability Zones
+
+**Feature Status:** Alpha since v1.12.
+
+Kubernetes v1.12 adds support for [Azure availability zones (AZ)](https://azure.microsoft.com/en-us/global-infrastructure/availability-zones/). Nodes in availability zone will be added with label `failure-domain.beta.kubernetes.io/zone=<region>-<AZ>` and topology-aware provisioning is added for Azure managed disks storage class.
+
+**TOC:**
+
+<!-- TOC -->
+
+- [Pre-requirements](#pre-requirements)
+- [Node labels](#node-labels)
+- [Load Balancer](#load-balancer)
+- [Managed Disks](#managed-disks)
+    - [StorageClass examples](#storageclass-examples)
+    - [PV examples](#pv-examples)
+- [Appendix](#appendix)
+- [Reference](#reference)
+
+<!-- /TOC -->
+
+## Pre-requirements
+
+Because only standard load balancer is supported with AZ, it is a prerequisite to enable AZ for the cluster. It should be configured in Azure cloud provider configure file (e.g. `/etc/kubernetes/azure.json`):
+
+```json
+{
+    "loadBalancerSku": "standard",
+    ...
+}
+```
+
+If topology-aware provisioning feature is used, feature gates `VolumeScheduling` and `DynamicProvisioningScheduling` should be enabled on master components (e.g. kube-apiserver, kube-controller-manager and kube-scheduler).
+
+## Node labels
+
+Both zoned and unzoned nodes are supported, but the value of node label `failure-domain.beta.kubernetes.io/zone` are different:
+
+- For zoned nodes, the value is `<region>-<AZ>`, e.g. `centralus-1`.
+- For unzoned nodes, the value is faultDomain, e.g. `0`.
+
+e.g.
+
+```yaml
+$ kubectl get nodes --show-labels
+NAME                STATUS    AGE   VERSION    LABELS
+kubernetes-node12   Ready     6m    v1.11      failure-domain.beta.kubernetes.io/region=centralus,failure-domain.beta.kubernetes.io/zone=centralus-1,...
+```
+
+## Load Balancer
+
+`loadBalancerSku` has been set to `standard` in cloud provider configure file, so standard load balancer and standard public IPs will be provisioned automatically for services with type `LoadBalancer`. Both load balancer and public IPs are zone redundant.
+
+## Managed Disks
+
+Zone-aware and topology-aware provisioning are supported for Azure managed disks. To support these features, a few options are added in AzureDisk storage class:
+
+- **zoned**: indicates whether new disks are provisioned with AZ. Default is true.
```

Comment:
Will this cause backwards compatibility issues for users that have StorageClasses before this change?

Reply:
No, azure supports both zoned and unzoned nodes together. If there's no zoned nodes in the cluster, then the disks will still be provisioned without zone.